tooling/minitest/src/tests/locks.rs

```diff
@@ -12,37 +12,45 @@ use crate::*;
 /// By making the critical section large (256 times around a loop)
 /// we get a high probability that the handover happened.
 fn lock_handover() {
-    let locals = [<()>::get_type(), <u32>::get_type()];
-
-    let b0 = block!(
-        storage_live(1),
-        assign(local(1), const_int::<u32>(0)),
-        lock_acquire(load(global::<u32>(0)), 1)
-    );
-    let b1 = block!(if_(eq(load(local(1)), const_int::<u32>(256)), 3, 2));
-    let b2 = block!(assign(local(1), add(load(local(1)), const_int::<u32>(1))), goto(1));
-    let b3 = block!(lock_release(load(global::<u32>(0)), 4));
-    let b4 = block!(return_());
-    let critical = function(Ret::Yes, 0, &locals, &[b0, b1, b2, b3, b4]);
-
-    let locals = [<u32>::get_type(), <()>::get_type()];
-
-    let b0 = block!(storage_live(0), storage_live(1), lock_create(global::<u32>(0), 1),);
-    let b1 = block!(spawn(fn_ptr(1), null(), local(0), 2));
-    let b2 = block!(call(2, &[], local(1), Some(3)));
-    let b3 = block!(join(load(local(0)), 4));
-    let b4 = block!(exit());
-    let main = function(Ret::No, 0, &locals, &[b0, b1, b2, b3, b4]);
-
-    let locals = [<()>::get_type(), <*const ()>::get_type()];
-
-    let b0 = block!(call(2, &[], local(0), Some(1)));
-    let b1 = block!(return_());
-    let second = function(Ret::Yes, 1, &locals, &[b0, b1]);
-
-    let globals = [global_int::<u32>()];
-
-    let p = program_with_globals(&[main, second, critical], &globals);
+    let mut p = ProgramBuilder::new();
+    let lock = p.global_int::<u32>();
+
+    let mut critical = p.function();
+    critical.ret::<()>();
+    let val = critical.local::<u32>();
+
+    critical.storage_live(val);
+    critical.assign(val, const_int(0u32));
+    critical.lock_acquire(load(lock));
+    critical.while_(ne(load(val), const_int(256_u32)), |f| {
+        f.assign(val, add(load(val), const_int(1u32)));
+    });
+    critical.lock_release(load(lock));
+    critical.return_();
+    let critical = p.finish_function(critical);
+
+    let mut second = p.function();
+
+    let mut main = p.function();
+    let thread_id = main.local::<u32>();
+    let unit = main.local::<()>();
+    main.storage_live(thread_id);
+    main.storage_live(unit);
+    main.lock_create(lock);
+    main.spawn(second.get_function_name(), null(), thread_id);
+    main.call(critical, &[], unit, Ret::Yes);
```

Comment:
Having to declare the `unit` here is annoying. What about a `call_ignoreret` function that
- declares a new local for the return value
- makes it live just before the call
- makes it dead just after the call

Reply:
Added `call_ignoreret`. It only works for functions with `()` return type (otherwise we would have to pass the return type), but this fits all current use cases.